vortex-array/src/array/primitive/mod.rs

```diff
@@ -191,3 +194,184 @@ impl<'a> Array<'a> {
 }
 
 impl EncodingCompression for PrimitiveEncoding {}
+
+impl ScalarSubtractFn for PrimitiveArray<'_> {
+    fn scalar_subtract(&self, to_subtract: &Scalar) -> VortexResult<OwnedArray> {
+        if !self.dtype().eq_ignore_nullability(to_subtract.dtype()) {
+            vortex_bail!(MismatchedTypes: self.dtype(), to_subtract.dtype())
+        }
+
+        let should_wrap = match self.validity() {
+            Validity::AllInvalid => {
+                return Ok(self.clone().into_array().to_static());
+            }
+            Validity::NonNullable | Validity::AllValid => false,
+            Validity::Array(_) => true,
+        };
+
+        let result = match to_subtract.dtype() {
+            DType::Int(..) => {
+                match_each_integer_ptype!(self.ptype(), |$T| {
+                    let to_subtract = $T::try_from(to_subtract)?;
+                    let maybe_min = self.statistics().compute_as_cast(Stat::Min);
+
+                    if let Some(min) = maybe_min {
+                        let min: $T = min;
+                        if let (min, true) = min.overflowing_sub(to_subtract) {
+                            vortex_bail!("Integer subtraction over/underflow: {}, {}", min, to_subtract)
+                        }
+                        if let Some(max) = self.statistics().compute_as_cast(Stat::Max) {
```

Comment:
I think you can treat max independently of min? If to_subtract < 0, then check max, >0 then check min, == 0 then can skip.

Reply:
You could add `is_negative()` `is_positive()` `is_zero()` function to PScalar

Reply:
I think we need to be careful here -- you mentioned earlier that we can have a max but no min. If we have a max, but no min, and max == 0, we could still underflow by subtracting a positive. 

Basically I think we need to check both for all nonsero to_subtract because of the optional nature of stats.